ts-reaktive-marshal-akka/src/main/java/com/tradeshift/reaktive/csv/CsvParser.java

```
@@ -0,0 +1,165 @@
+                        if (!buf.isEmpty()) {
+                            // emit the remaining buffer, unless we were in a quoted field and it's just the remaining quote
+                            if (state != State.QUOTED_FIELD || !buf.equals(String.valueOf(settings.getQuote()))) {
+                                emit(out, CsvEvent.text(settings.unescape(buf)));
+                            }
+                        }
+                        if (state == State.QUOTED_FIELD || state == State.UNQUOTED_FIELD || state == State.AFTER_QUOTED_FIELD) {
+                            emit(out, CsvEvent.endValue());
+                        }
+                        if (state == State.QUOTED_FIELD || state == State.UNQUOTED_FIELD || state == State.AFTER_QUOTED_FIELD) {
+                            emit(out, CsvEvent.endRecord());
+                        }
+                        completeStage();
+                    };
+                });
+            }
+            
+            private String parseAndEmit(String s) {
+                int p = (s.charAt(0) == BOM ? 1 : 0);
+                while (p < s.length()) {
+                    System.out.println("[" + p + "] " + state);
+                    switch(state) {
+                        case FIELD_START:
+                            if (s.charAt(p) == settings.getQuote()) {
+                                p++;
+                                state = State.QUOTED_FIELD;
+                                if (p > s.length()) {
+                                    return "";
+                                }
+                            } else if (s.charAt(p) == settings.getSeparator()) {
+                                // empty field
+                                emit(out, CsvEvent.endValue());
+                                p++;
+                            } else if (s.charAt(p) == '\n') {
+                                // end of record
+                                emit(out, CsvEvent.endValue());
```

In the case of having a csv record line this `"aaa";"bbb";"ccc"; CRLF ` I assume that the code will insert two "endValue" while parsing. I think you should either throw an error or avoid inserting the second one.